plugins/woocommerce-blocks/tests/e2e/tests/product-collection/register-product-collection-tester.block_theme.spec.ts

```diff
@@ -356,4 +356,83 @@ test.describe( 'Product Collection registration', () => {
 			await expect( previewButtonLocator ).toBeHidden();
 		} );
 	} );
+
+	test( 'Product picker should be shown when selected product is deleted', async ( {
+		pageObject,
+		admin,
+		editor,
+		requestUtils,
+	} ) => {
+		// Add a new test product to the database
+		let testProductId: number | null = null;
+		const newProduct = await requestUtils.rest( {
+			method: 'POST',
+			path: 'wc/v3/products',
+			data: {
+				name: 'A Test Product',
+				price: 10,
+			},
+		} );
+		testProductId = newProduct.id;
+
+		await admin.createNewPost();
+		await pageObject.insertProductCollection();
+		await pageObject.chooseCollectionInPost(
+			'myCustomCollectionWithProductContext'
+		);
+
+		// Verify that product picker is shown in Editor
+		const editorProductPicker = editor.canvas.locator(
+			SELECTORS.productPicker
+		);
+		await expect( editorProductPicker ).toBeVisible();
+
+		// Once a product is selected, the product picker should be hidden
+		await pageObject.chooseProductInEditorProductPickerIfAvailable(
+			editor.canvas,
+			'A Test Product'
+		);
+		await expect( editorProductPicker ).toBeHidden();
+
+		await editor.saveDraft();
+
+		// Delete the product
+		await requestUtils.rest( {
+			method: 'DELETE',
+			path: `wc/v3/products/${ testProductId }`,
+		} );
+
+		// Product picker should be shown in Editor
+		await admin.page.reload();
```

Comment:
Let's use the `page` fixture directly here and below. The `page` object is not always available through fixtures (private), so it might be confusing and prone to becoming unavailable. 

```suggestion
		await page.reload();
```